src/Atmos/Model/AtmosModel.jl

```diff
@@ -229,12 +228,95 @@ Set the value at the boundary to match the `init_state!` function. This is mainl
 struct InitStateBC <: BoundaryCondition
 end
 function boundarycondition!(m::AtmosModel{T,M,R,S,BC,IS}, stateP::Vars, diffP::Vars, auxP::Vars,
-    nM, stateM::Vars, diffM::Vars, auxM::Vars, bctype, t) where {T,M,R,S,BC <: InitStateBC,IS}
+    nM, stateM::Vars, diffM::Vars, auxM::Vars, bctype, t, _...) where {T,M,R,S,BC <: InitStateBC,IS}
   init_state!(m, stateP, auxP, auxP.coord, t)
 end
 
-function init_state!(m::AtmosModel, state::Vars, aux::Vars, coords, t)
-  m.init_state(state, aux, coords, t)
+function init_state!(m::AtmosModel, state::Vars, aux::Vars, coord, t)
+  m.init_state(state, aux, coord, t)
+end
+
+"""
+  DYCOMS_BC <: BoundaryCondition
+  Prescribes boundary conditions for Dynamics of Marine Stratocumulus Case
+"""
+struct DYCOMS_BC <: BoundaryCondition
+end
+function boundarycondition!(bl::AtmosModel{T,M,R,S,BC,IS}, stateP::Vars, diffP::Vars, auxP::Vars,
+    nM, stateM::Vars, diffM::Vars, auxM::Vars, bctype, t, state1::Vars, diff1::Vars, aux1::Vars) where {T,M,R,S,BC <: DYCOMS_BC,IS}
+    # stateM is the 𝐘⁻ state while stateP is the 𝐘⁺ state at an interface. 
+    # at the boundaries the ⁻, minus side states are the interior values
+    # state1 is 𝐘 at the first interior nodes relative to the bottom wall 
+    
+    # Get values from minus-side state
+    ρM = stateM.ρ 
+    UM, VM, WM = stateM.ρu
+    EM = stateM.ρe
+    QTM = stateM.moisture.ρq_tot
+    uM, vM, wM  = UM/ρM, VM/ρM, WM/ρM
+    q_totM = QTM/ρM
+    UnM = nM[1] * UM + nM[2] * VM + nM[3] * WM
+    
+    # Assign reflection wall boundaries (top wall)
+    stateP.ρu = SVector(UM - 2 * nM[1] * UnM, 
+                        VM - 2 * nM[2] * UnM,
+                        WM - 2 * nM[3] * UnM)
+
+    # Assign scalar values at the boundaries 
+    stateP.ρ = ρM
+    stateP.moisture.ρq_tot = QTM
+    # Assign diffusive fluxes at boundaries
+    diffP = diffM
+    xvert = auxM.coord[3]
+    
+    if xvert < 0.00001
```

Comment:
Why do we need this conditional?

Reply:
We do, unfortunately, for now. We need an elegant way to determine when we're at the ground element.

Reply:
If we use the `boundary` flag in the topology setup then we can assign specific indices to each `boundary` wall - this information is contained in `bctype` : default values for `bctype` are 1 (non-periodic) or 0 (periodic)

Reply:
@asraero is correct.  `bctype` is designed to be used to determine which boundary is the ground.  You should not need that conditional.

Reply:
Awesome

Reply:
if `bctype` indicates what physical boundary we are on (e.g. bottom wall, left wall, etc.) and not the `type` of boundary condition that is being applied, I suggest renaming`bctype` to something like `bdyposition`.

It would also help define it with a descriptive string rather than a number: 'BOTTOM_WALL', 'TOP_WALL', etc.

Reply:
`bctype` is an integer (which exists for each face) designed to be used to indicate the type of the boundary condition that is to be applied.

As @asraero indicated, you can set these numbers when defining your topology, for example see [[1]].  Currently, we support setting different numbers for the faces of the brick mesh or the top and bottom of the cubed sphere.  

I don't think we can use a string on the GPU.  I would suggest using an `Enum` to define your boundary condition types.  I am guessing this should go into the `AtmosModel`

[1]: https://github.com/climate-machine/CLIMA/blob/master/src/Mesh/Topologies.jl#L228-L230